static/app/components/groupPreviewTooltip/spanEvidencePreview.tsx

```diff
@@ -0,0 +1,158 @@
+import {Fragment, ReactChild, useEffect} from 'react';
+import styled from '@emotion/styled';
+
+import {SpanEvidenceKeyValueList} from 'sentry/components/events/interfaces/performance/spanEvidenceKeyValueList';
+import {getSpanInfoFromTransactionEvent} from 'sentry/components/events/interfaces/performance/utils';
+import {GroupPreviewHovercard} from 'sentry/components/groupPreviewTooltip/groupPreviewHovercard';
+import {useDelayedLoadingState} from 'sentry/components/groupPreviewTooltip/utils';
+import LoadingIndicator from 'sentry/components/loadingIndicator';
+import {t} from 'sentry/locale';
+import space from 'sentry/styles/space';
+import {EventTransaction} from 'sentry/types';
+import useApiRequests from 'sentry/utils/useApiRequests';
+import useOrganization from 'sentry/utils/useOrganization';
+
+type SpanEvidencePreviewProps = {
+  children: ReactChild;
+  eventId?: string;
+  groupId?: string;
+  projectSlug?: string;
+};
+
+type SpanEvidencePreviewBodyProps = {
+  endpointUrl: string;
+  onRequestBegin: () => void;
+  onRequestEnd: () => void;
+  onUnmount: () => void;
+};
+
+type Response = {
+  event: EventTransaction;
+};
+
+const makeGroupPreviewRequestUrl = ({
+  orgSlug,
+  eventId,
+  groupId,
+  projectSlug,
+}: {
+  orgSlug: string;
+  eventId?: string;
+  groupId?: string;
+  projectSlug?: string;
+}) => {
+  if (eventId && projectSlug) {
+    return `/projects/${orgSlug}/${projectSlug}/events/${eventId}/`;
+  }
+
+  if (groupId) {
+    return `/issues/${groupId}/events/latest/`;
+  }
+
+  return null;
+};
+
+const SpanEvidencePreviewBody = ({
+  endpointUrl,
+  onRequestBegin,
+  onRequestEnd,
+  onUnmount,
+}: SpanEvidencePreviewBodyProps) => {
+  const {data, isLoading, hasError} = useApiRequests<Response>({
+    endpoints: [['event', endpointUrl]],
+    onRequestError: onRequestEnd,
+    onRequestSuccess: onRequestEnd,
+  });
+
+  useEffect(() => {
+    onRequestBegin();
+
+    return () => {
+      onUnmount();
+    };
```

Comment:
Can you just 
```suggestion
    return onUnmount;
```

Have to be really careful here though. Any time `onRequestBegin` or `onUnmount` change it will re run this. So those MUST be useCallbacks or similar to maintain a stable identity.

It's a very foot-gunny API

Reply:
Yeah I usually wrap my cleanup functions for clarity but I went ahead and changed this

And yes! Those callbacks come from `useDelayedLoadingState()` which ensures they are stable